### DISCUSSION TITLE
How to change lang of element-plus in runtime

### DISCUSSION THREAD

**Discussion by barkleyhsiao** [element-plus/element-plus#4429] [Q&A]
Hi , how to change EP language to different one in runtime? I have setup the dropdown list to switch language, every thing is ok after change language except EP component like el-date-pick, it still state in language that before changes until refresh the page.
does it any way to change the language of EP in runtime after Application initiation is completed

**Comment by sxzz** [comment]
Could you please provide a minimum repo?

**Reply by barkleyhsiao** [reply]
In previous element-ui with Vue 2 , I could use below code to change element-ui language in runtime, I could not found the same function to change the language when the locale been change in EP
`import ElementLocale from 'element-ui/lib/locale' `
`import elzhTW from 'element-ui/lib/locale/lang/zh-TW'`
`import elEn from 'element-ui/lib/locale/lang/en'`
`if(this.$i18n.locale==='en'){`
`  ElementLocale.use(elEn)`
`}else{`
` ElementLocale.use(elzhTW)`
`}`

In EP with Vue 3 , I use below code to enable the i18n function, but could not found the way to achieve same goal. after I change the locale of i18n instance, the EP locale still remains the old.

vue-i18n initial part :
 
`import { createI18n } from "vue-i18n"`
`import elementEnLocale from "element-plus/lib/locale/lang/en"`
`import elementZhTWLocale from "element-plus/lib/locale/lang/zh-tw"`
`const messages = {`
`  en: {`
`    ...elementEnLocale,`
`  },`
`  "zh-tw": {`
`    ...elementZhTWLocale,`
`  },`
`}`

`const i18n = createI18n({`
`  legacy: false,`
`  globalInjection: true,`
`  locale: "zh-tw",`
`  fallbackLocale: "en",`
`  messages: messages,`
`  missingWarn: false,`
`})`
 
element-plus initial part:

`import ElementPlus from "element-plus"`
`locale = i18n.global.getLocaleMessage(i18n.global.locale.value)`
`app.use(ElementPlus, { size: "small", i18n: i18n.global.t, locale: locale })`
